test/bdd/agent_controller_steps.go

```diff
@@ -215,6 +214,43 @@ func (a *AgentWithControllerSteps) validateConnection(agentID, stateValue string
 		return fmt.Errorf("Expected state[%s] for agent[%s], but got[%s]", stateValue, agentID, response.Result.State)
 	}
 
+	// Also make sure new connection is available in list of connections for given agent
```

Comment:
which feature step is this function included in?

Reply:
While validating connection state after did-exchange, we already have one verification using getConnectionByID. This is just one more layer of verification, so that if something goes wrong with query connections, this bddtest step should fail.